src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { ReactElement, useCallback, useEffect } from 'react'
+import { CircularProgress, Typography } from '@mui/material'
+import { trackSafeAppOpenCount } from '@/services/safe-apps/track-app-usage-count'
+import { useSafeAppFromManifest } from '@/hooks/safe-apps/useSafeAppFromManifest'
+import { useSafeApps } from '@/hooks/safe-apps/useSafeApps'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { isSameUrl } from '@/utils/url'
+import { ThirdPartyCookiesWarning } from './ThirdPartyCookiesWarning'
+import useThirdPartyCookies from './useThirdPartyCookies'
+import useAppIsLoading from './useAppIsLoading'
+
+import css from './styles.module.css'
+
+export type TransactionParams = {
+  safeTxGas?: number
+}
+
+const URL_NOT_PROVIDED_ERROR = 'App url No provided or it is invalid.'
+
+type AppFrameProps = {
+  appUrl: string
+}
+
+const AppFrame = ({ appUrl }: AppFrameProps): ReactElement => {
+  const { safe } = useSafeInfo()
+  const { remoteSafeApps } = useSafeApps()
+  const { safeApp: safeAppFromManifest } = useSafeAppFromManifest(appUrl, safe.chainId)
+  const { thirdPartyCookiesDisabled, setThirdPartyCookiesDisabled } = useThirdPartyCookies()
+  const { iframeRef, appIsLoading, isLoadingSlow, setAppIsLoading } = useAppIsLoading()
+
+  const currentApp = remoteSafeApps?.filter((app: { url: string }) => app.url === appUrl)[0]
+
+  useEffect(() => {
+    if (!currentApp) return
+
+    trackSafeAppOpenCount(currentApp.id)
+  }, [currentApp])
+
+  const onIframeLoad = useCallback(() => {
+    const iframe = iframeRef.current
+    if (!iframe || !isSameUrl(iframe.src, appUrl as string)) {
+      return
+    }
+
+    setAppIsLoading(false)
+  }, [appUrl, iframeRef, setAppIsLoading])
+
+  useEffect(() => {
+    if (!appUrl) {
+      throw Error(URL_NOT_PROVIDED_ERROR)
+    }
+  }, [appUrl])
```

Comment:
How can this even be the case?

Reply:
yeah, just porting code from safe-react and the PR is still in Draft status but you are right and this is not possible as we have the same control in the corresponding next page. Going to remove it completely